<launch>
  <master auto="start"/>
  <param name="/use_sim_time" value="true"/>

  <node name="map_server" pkg="map_server" type="map_server" args="$(find mtg_sim_bringup)/worlds/stage_config/maps/tepper.yaml" respawn="false" >
    <remap from="/map" to="/sim_map"/>
  </node>
  
  <node pkg="mtg_sim_bringup" type="AprilTagSim.py" name="generate_april_tags_node" output="screen" />

  <group ns="mtg_agent_bringup_node">
    <node pkg="stage_ros_add_pose_and_crash" type="stageros" name="stageros" args="$(find mtg_sim_bringup)/worlds/stage_config/worlds/tepper.world" output="screen" respawn="false" >
      <param name="base_watchdog_timeout" value="0.2"/>
    </node>
  </group>

  <node pkg="tf2_ros" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 1 map agent_0/odom" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 1 map agent_1/odom" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="link3_broadcaster" args="0 0 0 0 0 0 1 map agent_2/odom" />
  
  <node pkg="tf2_ros" type="static_transform_publisher" name="link4_broadcaster" args="0 0 0 0 0 0 1 map agent_3/odom" />

  <node pkg="tf2_ros" type="static_transform_publisher" name="link5_broadcaster" args="0 0 0 0 0 0 1 map agent_4/odom" />

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find mtg_sim_bringup)/rviz/multi_robot.rviz">
  </node>
</launch>
